debug_toolbar/panels/sql/decoders.py

```diff
@@ -0,0 +1,26 @@
+import base64
+import json
+
+
+class DebugToolbarJSONDecoder(json.JSONDecoder):
+    """Custom JSON decoder that reconstructs binary data during parsing."""
+
+    def decode(self, s):
+        """Override decode to apply reconstruction after parsing."""
+        obj = super().decode(s)
+        return self._reconstruct_params(obj)
+
+    def _reconstruct_params(self, params):
+        """Reconstruct parameters, handling lists and dicts recursively."""
+        if isinstance(params, list):
+            return [self._reconstruct_params(param) for param in params]
+        elif isinstance(params, dict):
+            if "__djdt_binary__" in params:
+                return base64.b64decode(params["__djdt_binary__"])
+            else:
+                return {
+                    key: self._reconstruct_params(value)
+                    for key, value in params.items()
+                }
+        else:
+            return params
```

Comment:
```suggestion
        match params:
            case list():
                return [self._reconstruct_params(param) for param in params]
            case dict() if "__djdt_binary__" in params:
                return base64.b64decode(params["__djdt_binary__"])
            case dict():
                return {
                    key: self._reconstruct_params(value)
                    for key, value in params.items()
                }
            case _:
                return params
```
We don't need to use it, but match-case is an option here. What do you think @eduzen?